language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -0,0 +1,481 @@
+//! Translation from stackless Move bytecode to LLVM.
+//!
+//! Move is a stack machine and challenging to translate directly to LLVM. The
+//! `move_model` crate provides a translation of Move bytecode to "stackless
+//! bytecode", which is well-suited to further translation to LLVM.
+//!
+//! The structure of this module naturally mirrors both the Move model and LLVM
+//! sys, with a `GlobalContext` holding the Move `GlobalEnv` and the LLVM
+//! `Context`. Modules are translated through a `ModuleContext`, and functions a
+//! `FunctionContext`, each of which may accessed cached information from the
+//! parent context, all linked through lifetimes.
+//!
+//!
+//! # Lifetimes
+//!
+//! This module attempts to keep distinct lifetimes distinct to avoid
+//! a situation where they have be disentangled later. The structures
+//! contain two named lifetimes:
+//!
+//! - `'mm` - the lifetime of types stored inside the `move_model` `GlobalEnv`
+//! - `'up` - reference up the callstack to the higher-level context struct
+//!
+//! When constructing a new context the local lifetime that becomes `'up`
+//! is named `'this`.
+//!
+//! In general though this compiler does not need to be efficient at compile time -
+//! we can clone things when it makes managing lifetimes easier.
+
+use move_model::ast as mast;
+use move_model::model as mm;
+use move_model::ty as mty;
+use move_stackless_bytecode::stackless_bytecode as sbc;
+use std::collections::HashMap;
+use crate::stackless::extensions::*;
+use crate::stackless::llvm;
+
+#[derive(Copy, Clone)]
+pub enum Target {
+    Solana,
+}
+
+impl Target {
+    fn triple(&self) -> &'static str {
+        match self {
+            Target::Solana => "bpfel-unknown-unknown",
```

Comment:
If/when llvm.rs changes to SBF back-end, this should become `sbf-solana-solana`.